mmv1/third_party/terraform/resources/resource_compute_disk_async_replication.go.erb

```diff
@@ -0,0 +1,317 @@
+		if secondaryState, ok := diskStatus.ResourceStatus.AsyncSecondaryDisks[resourceName]; ok {
+			// Note this mi:ght be other than ACTIVE or STOPPED, but we wait for proper state
+			// on replication start/stop so it shouldnt affect Terraform
+			log.Printf("[DEBUG] Secondary disk %s is in state: %s", resourceName, secondaryState.State)
+			secondaryDisk["state"] = secondaryState.State
+		}
+		secondaryDisks = append(secondaryDisks, secondaryDisk)
+	}
+
+	log.Printf("[DEBUG] Secondary disks: %v", secondaryDisks)
+	if err = d.Set("secondary_disk", secondaryDisks); err != nil {
+		return fmt.Errorf("Error setting secondary_disk: %s", err)
+	}
+	d.SetId(resourceId)
+	return nil
+}
+
+func resourceDiskAsyncReplicationDelete(d *schema.ResourceData, meta interface{}) error {
+	clientCompute, err := getComputeClientForAsyncReplication(d, meta)
+	if err != nil {
+		return err
+	}
+
+	zonal, zv, rv, _, err := getDiskFromConfig(d.Get("primary_disk").(string), d, meta)
+	if err != nil {
+		return err
+	}
+
+	var replicationStopped bool = false
+	for _, secondaryDiskIter := range d.Get("secondary_disk").([]interface{}) {
+		secondaryDiskMap := secondaryDiskIter.(map[string]interface{})
+		secondaryDisk := secondaryDiskMap["disk"].(string)
+
+		_, _, _, resourceName, err := getDiskFromConfig(secondaryDisk, d, meta)
+		if err != nil {
+			return err
+		}
+
+		diskStatus, err := getComputeDiskStatus(clientCompute, zonal, zv, rv)
+		if err != nil {
+			return err
+		}
+
+		if diskStatus.ResourceStatus == nil {
+			continue
+		}
+
+		if secondaryState, ok := diskStatus.ResourceStatus.AsyncSecondaryDisks[resourceName]; ok {
+			if secondaryState.State != "STOPPED" {
+				replicationStopped = true
+				if zonal {
+					_, err = clientCompute.Disks.StopAsyncReplication(zv.Project, zv.Zone, zv.Name).Do()
+					if err != nil {
+						return err
+					}
+				} else {
+					_, err = clientCompute.RegionDisks.StopAsyncReplication(rv.Project, rv.Region, rv.Name).Do()
+					if err != nil {
+						return err
+					}
+				}
+				err = resource.Retry(time.Minute*time.Duration(5), func() *resource.RetryError {
+					diskStatus, err := getComputeDiskStatus(clientCompute, zonal, zv, rv)
+					if err != nil {
+						return resource.NonRetryableError(err)
+					}
+					if secondaryState, ok := diskStatus.ResourceStatus.AsyncSecondaryDisks[resourceName]; ok {
+						if secondaryState.State != "STOPPED" {
+							time.Sleep(5 * time.Second)
+							return resource.RetryableError(fmt.Errorf("secondary disk %s state (%s) is not STOPPED", secondaryDisk, secondaryState))
+						}
+						return nil
+					}
+					return resource.NonRetryableError(fmt.Errorf("secondary disk %s state not available", secondaryDisk))
+				})
+				if err != nil {
+					return err
+				}
+			}
+		} else {
+			return fmt.Errorf("could not find secondary disk: %s", secondaryDisk)
+		}
+	}
+	if replicationStopped {
+		// Allow the replication to quiescence
+		time.Sleep(5000 * time.Millisecond)
+	}
```

Comment:
This seems not useful? Replication has already been stopped as the logic above already confirms